Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -183,6 +187,50 @@ inline void ClearPath() {
         renderTarget->GetFactory(&factory);
         return factory;
     }
+
+    HRESULT Clip(CGPathDrawingMode pathMode) {
+        if (!HasPath()) {
+            // Clipping to nothing is okay.
+            return S_OK;
+        }
+
+        auto& state = CurrentGState();
+
+        ComPtr<ID2D1Geometry> additionalClippingGeometry;
+        RETURN_IF_FAILED(_CGPathGetGeometry(Path(), &additionalClippingGeometry));
+        ClearPath();
+
+        D2D1_FILL_MODE d2dFillMode = (pathMode & kCGPathEOFill) == kCGPathEOFill ? D2D1_FILL_MODE_ALTERNATE : D2D1_FILL_MODE_WINDING;
+
+        if (!state.clippingGeometry) {
+            // If we don't have a clipping geometry, we are free to take this one wholesale (after EO/Winding conversion.)
+            ComPtr<ID2D1Geometry> convertedClippingGeometry;
+            RETURN_IF_FAILED(_CGConvertD2DGeometryToFillMode(additionalClippingGeometry.Get(), d2dFillMode, &convertedClippingGeometry));
```

Comment:
Oh! Yes. This got muddled in the condition flip AJ suggested.

---
In reply to: [87893114](https://github.com/Microsoft/WinObjC/pull/1368#discussion_r87893114) [](ancestors = 87893114)